src/main/java/com/dnastack/gatekeeper/routing/GatekeeperRequestRouter.java

```diff
@@ -88,9 +89,15 @@ private String choosePrefixBasedOnAuth(HttpServletRequest request, HttpServletRe
         Jws<Claims> jws;
         try {
             jws = jwtParser.parseClaimsJws(authToken);
+            
             log.info("Validated signature of inbound token {}", jws);
 
             final Claims claims = jws.getBody();
+            Date expiry = claims.getExpiration();
```

Comment:
I think `jwtParser.parseClaimsJws()` throws an exception on an expired token. If so, this new code won't get executed.

It would be worth adding a unit test that injects a JwtParser with a known key and then feeds in an expired token to the `choosePrefixBasedOnAuth()` method.

Reply:
I've changed it to return insufficient-credentials if it catches ExpiredJwtException, will write the unit test over the weekend.